include/swift/AST/DeclContext.h

```diff
@@ -619,6 +619,12 @@ class IterableDeclContext {
   /// Retrieve the set of members in this context.
   DeclRange getMembers() const;
 
+  /// Return either the set of members (that might be types) or an empty
+  /// DeclRange (if this is a context that can't have type members). Caller
+  /// still has to filter for types, but this can avoid loading members
+  /// unnecessarily.
+  DeclRange getMembersThatMightBeTypes() const;
```

Comment:
This is a weird API. If it's going to vend just the types within the content, please have it produce some range of `TypeDecl` nodes so the clients can be cleaner. Returning a `SmallVector<TypeDecl *, 4>` would be fine, for example.

Reply:
I agree. It's fine if the implementation is not always as optimal as possible, but it should still do it's own filtering in that case.

Reply:
Ok. I was aiming for minimal disruption & absence of new allocations, but that works too and I guess it removes some noise in the clients.